roles/openshift_registry/tasks/main.yml

```diff
@@ -2,10 +2,14 @@
 - set_fact: _oreg_images="--images={{ oreg_url|quote }}"
   when: oreg_url is defined
 
+- set_fact: _oreg_images="--images='registry.access.redhat.com/openshift3/ose-${component}:${version}'"
+  when: deployment_type == 'online'
+
```

Comment:
openshift.master.registry_url ... That's a good idea. I didn't know that existed already in openshift_facts. Then I wouldn't need the 'when', like you said. And I wouldn't need to specify the online registry url separately.

I'm not sure what you mean about dropping the defaults in openshift_facts. Could you explain how you'd like it to work?

My current understanding is, there are defaults in openshift_facts that set registry_url based on the deployment type (online, origin, enterprise). It seems like a good idea to have those options set for us. Then, if a user wants to specify a different registry url, they can overwrite registry_url defaults by specifying oreg_url=something in their inventory file. (This is what I mean:)

https://github.com/openshift/openshift-ansible/blob/master/roles/openshift_master/tasks/main.yml#L48

In that case, I think I should be able to get rid of both those set_fact's listed above, and replace it with the `openshift.master.registry_url` you mentioned:

`- set_fact: _oreg_images="--images={{ openshift.master.registry_url }}"`